contracts/upgradeable/MultiCallerUpgradeable.sol

```diff
@@ -46,6 +46,29 @@ contract MultiCallerUpgradeable {
         //slither-disable-end calls-loop
     }
 
+    function multicallRevertable(bytes[] calldata data) external returns (bytes[] memory results) {
+        _validateMulticallData(data);
+
+        uint256 dataLength = data.length;
+        results = new bytes[](dataLength);
+
+        //slither-disable-start calls-loop
+        for (uint256 i = 0; i < dataLength; i++) {
+            // Typically, implementation contracts used in the upgradeable proxy pattern shouldn't call `delegatecall`
+            // because it could allow a malicious actor to call this implementation contract directly (rather than
+            // through a proxy contract) and then selfdestruct() the contract, thereby freezing the upgradeable
+            // proxy. However, since we're only delegatecall-ing into this contract, then we can consider this
+            // use of delegatecall() safe.
+
+            //slither-disable-start low-level-calls
+            /// @custom:oz-upgrades-unsafe-allow delegatecall
+            (, bytes memory result) = address(this).delegatecall(data[i]);
```

Comment:
nit: it's probably a good idea to tell the caller which calls succeeded and which errored, so maybe we should return an array of structs with a boolean _and_ bytes?